File: projects/NativeQueueSample/test/NativeQueueSample.Test.Unit/Test1.cpp
//-----------------------------------------------------------------------
// <copyright file="Test1.cpp" company="Brian Rogers">
// Copyright (c) Brian Rogers. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

#include <ppl.h>
#include "Class1.h"
#include "CppUnitTest.h"

using namespace concurrency;
using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace std;

namespace NativeQueueSample
{		
    TEST_CLASS(Test1)
    {
    public:
        TEST_METHOD(Should_return_name_after_completing_sync)
        {
            Class1 c(L"MyName");

            task<wstring> task = c.DoAsync();

            Assert::IsTrue(task.is_done());
            Assert::AreEqual(wstring(L"MyName"), task.get());
        }
    };
}

File: projects/EventHandlerSample/test/EventHandlerSample.Test.Unit/Test1.cs
//-----------------------------------------------------------------------
// <copyright file="Test1.cs" company="Brian Rogers">
// Copyright (c) Brian Rogers. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

namespace EventHandlerSample.Test.Unit
{
    using System.Threading.Tasks;
    using FluentAssertions;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class Test1
    {
        public Test1()
        {
        }

        [TestMethod]
        public void ShouldReturnNameAfterCompletingSync()
        {
            Class1 c = new Class1("MyName");
            
            Task<string> task = c.DoAsync();

            task.IsCompleted.Should().BeTrue();
            task.Result.Should().Be("MyName");
        }
    }
}
